### ISSUE TITLE
How to use wifi adapter using otg on termux?

### ISSUE THREAD

**Issue by NightShade-Reaper** [nohajc/termux-adb#20]
I am currently using the latest version of this software. 

I am seeking for some help as i don't know how to use it .
I want to use a wifi adapter on termux using otg .
My phone is non-rooted. 

Thanks for your assistance

**Comment by nohajc** [comment]
I'm not sure I understand what you're trying to achieve. Termux-adb lets you connect from one Android device to another via USB using the adb protocol (i.e. one phone runs the adb server, it can enumerate all connected Android devices and attach to their shell, install apps, show logs, etc.).

There's also adb over Wi-Fi (or any LAN in general) that can be enabled in developer settings (it reverts on each boots though). This one you can use with normal adb without root.

Are you connecting a USB Wi-Fi adapter to your device? Even if Termux could see it (thanks to termux-usb which is used under the hood), I doubt you could use it as a network interface if Android itself doesn't detect it. That would require kernel-level support.

Basically, termux-adb works only because adb is implemented entirely in user space.

**Comment by wildxxxx** [comment]
hello! I wanted to make a usb pass-through in qemu on termux, but the output of "info usbhost" says that it uses "port 0" and "Speed ? Mb/s". Which is why I can't use it in qemu. What should I do about it? Thank you.

**Comment by wildxxxx** [comment]
> I'm not sure I understand what you're trying to achieve. Termux-adb lets you connect from one Android device to another via USB using the adb protocol (i.e. one phone runs the adb server, it can enumerate all connected Android devices and attach to their shell, install apps, show logs, etc.).
> 
> There's also adb over Wi-Fi (or any LAN in general) that can be enabled in developer settings (it reverts on each boot though). This one you can use with normal adb without root.
> 
> Are you connecting a USB Wi-Fi adapter to your device? Even if Termux could see it (thanks to termux-usb which is also used by termux-adb under the hood), I doubt you could use it as a network interface if Android itself doesn't detect it. That would require kernel-level support.
> 
> Basically, termux-adb only works because adb is implemented entirely in user space.

hello! I wanted to make a usb pass-through in qemu on termux, but the output of "info usbhost" says that it uses "port 0" and "Speed ? Mb/s". Which is why I can't use it in qemu. What should I do about it? Thank you.

**Comment by nohajc** [comment]
Interesting question, but I never tried qemu on Termux. Could be that the USB interface exposed by Termux is insufficient for this use case. You could try asking the Termux devs about this.
